### DISCUSSION TITLE
About highlighting of `extraConfigLua`

### DISCUSSION THREAD

**Discussion by refaelsh** [nix-community/nixvim#2093] [Q&A] [answered]
I saw somewhere in the docs that it is possible to set highlighting for `extraConfigLua` blocks in configuration.nix.
Now I can find it. Can somebody please point me the right direction? Thanks!

**Comment by traxys** [comment, marked as answer]
Here is the option: https://nix-community.github.io/nixvim/plugins/treesitter/index.html#pluginstreesitternixviminjections !

**Reply by MattSturgeon** [reply]
Another thing you can do is "tag" the string by having a `/* lua */` or `# lua` comment _immediately_ before it. Treesitter will pick up on this and highlight it as appropriate.

E.g.

```nix
{
  extraConfigLua = /* lua */ ''
    -- will be highlighted
  '';
  extraConfigLuaPost = # lua
    "-- so will this";
}
```

**Reply by refaelsh** [reply]
Really cool! I just tried it and it works. Thank you :-)

**Reply by refaelsh** [reply]
> Here is the option: https://nix-community.github.io/nixvim/plugins/treesitter/index.html#pluginstreesitternixviminjections !

1. This option says that the default is true - but it does not work.
2. I set it to true - and it still does not work.

I think it's a bug. What do you think?

**Reply by khaneliman** [reply]
Do you have `plugins.treesitter.settings.highlight.enable = true;` ? The injections only do something when you tell treesitter to use it's highlight functionality. Although, if you can see the comment language hinting but not the injections, then you must have highlighting enabled and the queries aren't getting picked up?

**Reply by MattSturgeon** [reply]
IIRC the injections only apply to `__raw` and `mkRaw`? Although @traxys wrote them, so he will have a better idea.

**Reply by khaneliman** [reply]
Nah, they work on `(#match? @_path "(^(extraConfigLua(Pre|Post)?|__raw))$"))` and `(#match? @_path "(^extraConfigVim(Pre|Post)?)$"))`

**Reply by khaneliman** [reply]
IIRC though, he had to update them with the new neovim version bump? Is it possible you're running neovim 0.9 ?

**Reply by MattSturgeon** [reply]
nixos-24.05 is still using nvim 0.9, however if the user is on 24.05 they should be using our matching branch, which has an older [injections file](https://github.com/nix-community/nixvim/blob/764b89aa14543da7266719757cfcf0cce8c1679f/plugins/languages/treesitter/treesitter.nix#L179-L198).

Looks like it still had `extraConfigLua` support back then, though...

**Reply by khaneliman** [reply]
Unstable nixpkgs with stable nixvim branch and an overwritten follows ? :P Just guessing why it wouldn't work. It's been working for me since that commit for updating the injections file.

**Reply by refaelsh** [reply]
> Do you have plugins.treesitter.settings.highlight.enable = true; ?

Yes.

> however if the user is on 24.05 they should be using our matching branch

I am on unstable.

**Reply by traxys** [reply]
Do you have the lua & nix parsers? I'm not really sure what happens here, as it seems to work for others?

Note that the injection is quite picky, you can't have anything between `extraConfigLua = ''<code>''`.

**Reply by refaelsh** [reply]
> Note that the injection is quite picky, you can't have anything between extraConfigLua = ''<code>''.

This is interesting. Let me play with it for a bit.

**Reply by refaelsh** [reply]
Ok, you were right, its the `quite picky` nature of it. I played with it, added spaces, removed tabs, moved lines and etc, and all of a sudden it started working.
